app/controllers/admin/export_domain_blocks_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Admin
+  class ExportDomainBlocksController < BaseController
+    include AdminExportControllerConcern
+
+    ROWS_PROCESSING_LIMIT = 20_000
+
+    def new
+      authorize :domain_block, :create?
+    end
+
+    def export
+      authorize :instance, :index?
+      send_export_file
+    end
+
+    def import
+      authorize :domain_block, :create?
+      @import = Admin::Import.new(import_params)
+      parse_import_data!(export_headers)
+
+      @data.take(ROWS_PROCESSING_LIMIT).each do |row|
+        domain = row['#domain'].strip
+        next if DomainBlock.rule_for(domain).present?
+
+        domain_block = DomainBlock.new(domain: domain,
+                                       severity: row['#severity'].strip,
+                                       reject_media: row['#reject_media'].strip,
+                                       reject_reports: row['#reject_reports'].strip,
+                                       public_comment: row['#public_comment'].strip,
+                                       obfuscate: row['#obfuscate'].strip)
+        if domain_block.save
+          DomainBlockWorker.perform_async(domain_block.id)
+          log_action :create, domain_block
+        end
+      end
```

Comment:
`ActionController::ParameterMissing` should probably be rescued so that an helpful error message is given when no file is uploaded.